client/client.go

```diff
@@ -72,12 +73,22 @@ var (
 
 	// ErrInitializationNotReceived occurs when the first response of the watch stream is not received.
 	ErrInitializationNotReceived = errors.New("initialization is not received")
+
+	// ErrAlreadySubscribed occurs when the client is already subscribed to the document.
+	ErrAlreadySubscribed = errors.New("already subscribed")
 )
 
 // Attachment represents the document attached.
 type Attachment struct {
 	doc   *document.Document
 	docID types.ID
+
+	// TODO(krapie): assuming that a client do not open multiple subscriptions for the same document.
+	isSubscribed atomic.Bool
```

Comment:
This comment is `TODO`, so how about writing about what we think will happen in the future?